packages/server/src/games.ts

```diff
@@ -186,12 +195,21 @@ export async function handleMoveAndTime(
     }
   }
 
-  gamesCollection()
-    .updateOne(
-      { _id: new ObjectId(game_id) },
-      { $push: { moves: moves }, $set: { time_control: timeControl } },
+  const moveUpdateResult = await gamesCollection()
```

Comment:
This is also used in `timeout.ts`, where i wonder if we should change it s.t. the timeout move is always pushed on top, or not.

Reply:
That's a good callout.

For  sequential, where the same user who plays is the one who times out, I think it's fine for the timeout to fail.

For the case of simultaneous moves, we'd want the timeout event to get pushed on.

Reply:
Do you think it is ok to leave it as is for now? Currently I don't see many games of parallel variants played with time control at all (also, I assume the race condition with timeout would be pretty rare anyway).